test/e2e/network_benchmark_test.go

```diff
@@ -31,21 +36,40 @@ func BenchmarkE2ENetwork(b *testing.B) {
 
 				signerKeyPair := keys.Ed25519KeyPairForTests(5)
 				targetAddress := builders.AddressForEd25519SignerForTests(6)
-				transfer := builders.TransferTransaction().WithEd25519Signer(signerKeyPair).WithAmountAndTargetAddress(uint64(rand.Intn(i)), targetAddress).Builder()
+				amount := uint64(rand.Intn(10))
+
+				transfer := builders.TransferTransaction().WithEd25519Signer(signerKeyPair).WithAmountAndTargetAddress(amount, targetAddress).Builder()
 				_, err := h.sendTransaction(transfer)
+
 				if err != nil {
-					fmt.Printf("error sending transaction %s\n", err)
+					t.Logf("error sending transaction %s\n", err)
 				}
 			}()
-
+		} else {
+			t.Logf("error %s\n", err)
 		}
-
 	}
 
-	res, _ := http.Get(h.absoluteUrlFor("/metrics"))
-	bytes, _ := ioutil.ReadAll(res.Body)
-	println(string(bytes))
-
 	wg.Wait()
 
-}
\ No newline at end of file
+	var m metrics
+
+	require.True(t, test.Eventually(1*time.Minute, func() bool {
+		m = h.getMetrics()
+		return m != nil
+	}), "could not retrieve metrics")
+
+	txCount := m["TransactionPool.CommittedPool.TransactionCount"]["Value"].(float64)
+
+	expectedNumberOfTx := float64((100 - config.acceptableFailureRate) / 100 * config.numberOfTransactions)
+
+	require.Condition(t, func() (success bool) {
```

Comment:
why do you need a named return argument?

Reply:
I doesn't have to have a name, but I think it improves readability a bit.